chia/rpc/wallet_rpc_api.py

```diff
@@ -1162,14 +1278,179 @@ async def did_get_information_needed_for_recovery(self, request):
             "backup_dids": did_wallet.did_info.backup_ids,
         }
 
+    async def did_get_current_coin_info(self, request):
+        wallet_id = int(request["wallet_id"])
+        did_wallet: DIDWallet = self.service.wallet_state_manager.wallets[wallet_id]
+        my_did = did_wallet.get_my_DID()
+        did_coin_threeple = await did_wallet.get_info_for_recovery()
+        assert my_did is not None
+        assert did_coin_threeple is not None
+        return {
+            "success": True,
+            "wallet_id": wallet_id,
+            "my_did": my_did,
+            "did_parent": did_coin_threeple[0],
+            "did_innerpuz": did_coin_threeple[1],
+            "did_amount": did_coin_threeple[2],
+        }
+
     async def did_create_backup_file(self, request):
+        wallet_id = int(request["wallet_id"])
+        did_wallet: DIDWallet = self.service.wallet_state_manager.wallets[wallet_id]
+        return {"wallet_id": wallet_id, "success": True, "backup_data": did_wallet.create_backup()}
+
+    async def did_transfer_did(self, request):
+        assert self.service.wallet_state_manager is not None
+        if await self.service.wallet_state_manager.synced() is False:
+            raise ValueError("Wallet needs to be fully synced.")
+        wallet_id = int(request["wallet_id"])
+        did_wallet: DIDWallet = self.service.wallet_state_manager.wallets[wallet_id]
+        puzzle_hash: bytes32 = decode_puzzle_hash(request["inner_address"])
+        async with self.service.wallet_state_manager.lock:
+            txs: TransactionRecord = await did_wallet.transfer_did(
+                puzzle_hash, uint64(request.get("fee", 0)), request.get("with_recovery_info", True)
+            )
+
+        return {
+            "success": True,
+            "transaction": txs.to_json_dict_convenience(self.service.config),
+            "transaction_id": txs.name,
+        }
+
+    ##########################################################################################
+    # NFT Wallet
+    ##########################################################################################
+
+    async def nft_mint_nft(self, request) -> Dict:
+        log.debug("Got minting RPC request: %s", request)
+        wallet_id = uint32(request["wallet_id"])
+        assert self.service.wallet_state_manager
+        nft_wallet: NFTWallet = self.service.wallet_state_manager.wallets[wallet_id]
+        address = request.get("artist_address")
+        if isinstance(address, str):
```

Comment:
why allow many formats, better to just allow one format and enforce it consistently

Reply:
this is to be backwards compatible with other minting/creating methods